api/api_controller.go

```diff
@@ -1138,25 +1148,37 @@ func (c *Controller) ObjectsListObjectsHandler() objects.ListObjectsHandler {
 				WithPayload(responseError("error while listing objects: %s", err))
 		}
 
+		repo, err := cataloger.GetRepository(c.Context(), params.Repository)
+		if err != nil {
+			return objects.NewStatObjectDefault(http.StatusInternalServerError).WithPayload(responseErrorFrom(err))
+		}
+
 		objList := make([]*models.ObjectStats, len(res))
 		var lastID string
 		for i, entry := range res {
+			qk, err := block.ResolveNamespace(repo.StorageNamespace, entry.PhysicalAddress)
+			if err != nil {
+				return objects.NewStatObjectDefault(http.StatusInternalServerError).WithPayload(responseErrorFrom(err))
+			}
+
 			if entry.CommonLevel {
 				objList[i] = &models.ObjectStats{
-					Path:     entry.Path,
-					PathType: models.ObjectStatsPathTypeCommonPrefix,
+					Path:            entry.Path,
+					PhysicalAddress: qk.Format(),
```

Comment:
This adds a physical address to common prefixes as well, which is a little confusing (it simply returns the storage namespace of the repository). This could be misleading since that prefix might contain objects that don't belong in the storage namespace.